src/Microsoft.AspNetCore.Authentication.Facebook/FacebookHandler.cs

```diff
@@ -31,7 +32,7 @@ public FacebookHandler(HttpClient httpClient)
             }
             if (Options.Fields.Count > 0)
             {
-                endpoint = QueryHelpers.AddQueryString(endpoint, "fields", string.Join(",", Options.Fields));
+                endpoint = QueryHelpers.AddQueryString(endpoint, "fields", string.Join(",", Options.Fields.Distinct()));
```

Comment:
If you're going to do options validation like calling `Distinct()` it should be done once in the middleware constructor, not on every request.